services/spar/test-integration/Util.hs

```diff
@@ -308,36 +325,55 @@ shouldRespondWith action proper = do
 -- envit :: Example (r -> m a) => String -> ReaderT r m a -> SpecWith (Arg (r -> m a))
 -- envit msg action = it msg $ \env -> action `runReaderT` env
 
-call :: Http a -> ReaderT TestEnv IO a
+call :: (MonadIO m, MonadReader TestEnv m) => Http a -> m a
 call req = ask >>= \env -> liftIO $ runHttpT (env ^. teMgr) req
 
 ping :: (Request -> Request) -> Http ()
 ping req = void . get $ req . path "/i/status" . expect2xx
 
 
-createTestIdP :: (HasCallStack, MonadReader TestEnv m, MonadIO m) => m (UserId, TeamId, SAML.IdPId)
-createTestIdP = do
+-- | Create a cloned 'NewIdP' corresponding to the mock idp from the config and suitable for
+-- registering with spar.  The issuer id can be used to do this several times without getting the
+-- error that this issuer is already used for another team.
+makeTestNewIdP :: (HasCallStack, MonadReader TestEnv m, MonadIO m) => UUID -> m NewIdP
+makeTestNewIdP issuerid = do
   env <- ask
-  liftIO . runHttpT (env ^. teMgr) $ do
-    (uid, tid) <- createUserWithTeam (env ^. teBrig) (env ^. teGalley)
-    (uid, tid,) . (^. SAML.idpId) <$> callIdpCreate (env ^. teSpar) (Just uid) sampleIdP
+  let Endpoint ephost (cs . show -> epport) = env ^. teTstOpts . to cfgMockIdp
+      mkurl = either (error . show) id . SAML.parseURI' . (("http://" <> ephost <> ":" <> epport) <>)
```

Comment:
If you're doing it like this then leave a comment in the config file saying that the host URL must not have a slash